test/hotspot/jtreg/runtime/os/TestTimerSlack.java

```diff
@@ -0,0 +1,128 @@
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+
+import jdk.test.lib.process.ProcessTools;
+import jdk.test.lib.process.OutputAnalyzer;
+
+/**
+ * @test
+ * @summary Check that timer slack options work
+ * @requires os.family == "linux"
+ * @requires vm.flagless
+ * @library /test/lib
+ * @run driver TestTimerSlack
+ */
+public class TestTimerSlack {
+
+    public static void main(String[] args) throws Exception {
+        int defaultSlack;
+
+        // Check the timer slack value is good by default
+        {
+            ProcessBuilder pb = ProcessTools.createJavaProcessBuilder("-Xlog:os+thread",
+                                                      "TestTimerSlack$Hello");
+
+            OutputAnalyzer output = new OutputAnalyzer(pb.start());
+            output.shouldHaveExitValue(0);
+            output.shouldContain("timer slack:");
+
+            defaultSlack = parseSlackValue(output);
+
+            if (defaultSlack == 0) {
+                output.reportDiagnosticSummary();
+                throw new IllegalStateException("Default slack value (" + defaultSlack + ") is unexpected");
+            }
+        }
+
+        // Check the timer slack value is still default when explicitly disabled/default
+        for (int slack : new int[] {-1, 0}) {
+            ProcessBuilder pb = ProcessTools.createJavaProcessBuilder("-Xlog:os+thread",
+                                                      "-XX:+UnlockExperimentalVMOptions",
+                                                      "-XX:TimerSlack=" + slack,
+                                                      "TestTimerSlack$Hello");
+
+            OutputAnalyzer output = new OutputAnalyzer(pb.start());
+            output.shouldHaveExitValue(0);
+            output.shouldContain("timer slack:");
+
+            int disabledSlack = parseSlackValue(output);
+            if (disabledSlack != defaultSlack) {
+                output.reportDiagnosticSummary();
+                throw new IllegalStateException("Disabled slack value (" + disabledSlack +
+                    ") is not default (" + defaultSlack + ") when requested (" + slack + ")");
+            }
+        }
+
+        // Check the timer slack value is accepted by all threads
```

Comment:
I don't see anything thread related in the logic that follows. I expected to see the test program create threads and read back their time-slack values.

Reply:
Right. I was mostly concentrating on main thread. New commit spawns a few threads explicitly. Alas, we cannot check if those are the same threads, because our printout happens on thread creation path, when `setName` did not yet have a chance to set the thread name.